### ISSUE TITLE
Missing validation for forward creation?

### ISSUE THREAD

**Issue by davidebeatrici** [postfixadmin/postfixadmin#572]
I accidentally created a few forwards without specifying the domain, resulting in the following:

![](https://user-images.githubusercontent.com/5897523/144722492-8a3cc803-f6cf-4438-8705-6ab65c75d644.png)

I'm unable to edit 3 out of 4 entries. Crafting the edit link and submitting changes:

![](https://user-images.githubusercontent.com/5897523/144722579-61053fb3-ae25-4fe6-94c3-c6588623490d.png)

If I log in as superadmin I'm able to edit all forwards.

**Comment by DavidGoodwin** [comment]
I wouldn't expect the domain omission to be the problem.

**Comment by DavidGoodwin** [comment]
the code implies your user doesn't have permission to edit the alias; does it have permission to modify the domain in question?

**Comment by DavidGoodwin** [comment]
https://github.com/postfixadmin/postfixadmin/blob/master/model/PFAHandler.php#L432

**Comment by davidebeatrici** [comment]
Yes, I created the forwards while logged in as the user that is unable to edit them.

**Comment by DavidGoodwin** [comment]
as a user, if i try to add a non-fully-qualified foward, i see 

![image](https://user-images.githubusercontent.com/203929/145094201-76cb10c0-4341-4036-b6c9-84fda37390b7.png)

**Comment by DavidGoodwin** [comment]
when I login as an admin for a specific domain (in my case, just 'example.com' - so NOT a superadmin) - I see the same - where I can't edit e.g default aliases that were added when the domain was created.

**Comment by DavidGoodwin** [comment]
![image](https://user-images.githubusercontent.com/203929/145095743-be3f9f27-0d35-4af4-8faa-d4670bd67fc8.png)

**Comment by DavidGoodwin** [comment]
So at least for me, if i'm not a super-admin and the alias target is one of the ones defined in `special_aliases' ( https://github.com/postfixadmin/postfixadmin/blob/master/config.inc.php#L268 ) then edit permission is removed.

**Comment by DavidGoodwin** [comment]
the code that's turning off editability is here - https://github.com/postfixadmin/postfixadmin/blob/52ed517f35fc2bfde193fad94ddd07d2e5cd4c8d/model/AliasHandler.php#L346

**Comment by davidebeatrici** [comment]
> as a user, if i try to add a non-fully-qualified foward, i see
> 
> ![image](https://user-images.githubusercontent.com/203929/145094201-76cb10c0-4341-4036-b6c9-84fda37390b7.png)

Looks like a different page compared to the admin one. Is perhaps address validation only implemented for the user one?

> So at least for me, if i'm not a super-admin and the alias target is one of the ones defined in `special_aliases' ( [master/config.inc.php#L268](https://github.com/postfixadmin/postfixadmin/blob/master/config.inc.php?rgh-link-date=2021-12-07T20%3A03%3A45Z#L268) ) then edit permission is removed.

I unchecked the option to automatically create default aliases when I added the domain as superadmin.

As default aliases are only supposed to be modifiable by a superadmin (unless `special_alias_control` is enabled), I would expect domain admins not to be able to create them in the first place.
